crates/toolchain/edsl-kernel/src/generate_edsl.rs

```diff
@@ -0,0 +1,98 @@
+use crate::parse_kernel::ParsedKernel;
+
+const MAIN_PLACEHOLDER: &str = "<...>";
+const OUTSIDE_PLACEHOLDER: &str = "<===>";
+const EDSL_TEMPLATE: &str = "
+use openvm_native_compiler::{
+    asm::{AsmBuilder, AsmCompiler},
+    conversion::{convert_program, CompilerOptions},
+    ir::{Array, Ext, Felt},
+};
+use p3_baby_bear::BabyBear;
+use p3_field::{extension::BinomialExtensionField, AbstractField};
+
+type F = BabyBear;
+type EF = BinomialExtensionField<BabyBear, 4>;
+
+<===>
+
+fn main() {
+    let mut builder = AsmBuilder::<F, EF>::default();
+
+<...>
+
+    let mut compiler = AsmCompiler::new(1);
+    compiler.build(builder.operations);
+    let asm_code = compiler.code();
+    let program = convert_program::<F, EF>(asm_code, CompilerOptions::default());
+    println!(\"{}\", program);a stru
+}
+";
+
+pub const CONFIG: &str = "
+[package]
+name = \"root\"
+version = \"0.1.0\"
+edition = \"2021\"
+
+[dependencies]
+openvm-native-compiler = { path = \"../../../../extensions/native/compiler\" }
```

Comment:
we need a way to have dependency versions consistent with the workspace without hardcoding

Reply:
harcoding path is also not good